pyblish/lib.py

```diff
@@ -53,10 +53,31 @@ def emit(self, record):
             self.records.append(record)
 
 
-def extract_traceback(exception):
-    """Inject current traceback and store in exception"""
+def extract_traceback(exception, plugin):
+    """Inject current traceback and store in exception.traceback.
+
+    Also storing the formatted traceback on exception.formtatted_traceback.
+
+    Arguments:
+        exception (Exception): Exception object
+        plugin (pyblish.api.Plugin): Plugin class that caused the exception.
+            This is necessary to inject the correct file path in the traceback.
+            If plugins are registered through `api.plugin.discover`, they only
+            show "<string>" instead of the actual source file.
+    """
     exc_type, exc_value, exc_traceback = sys.exc_info()
     exception.traceback = traceback.extract_tb(exc_traceback)[-1]
+
+    formatted_traceback = ''.join(traceback.format_exception(
+        exc_type, exc_value, exc_traceback))
+    if 'File "<string>", line ' in formatted_traceback:
+        fname, lineno, func, msg = exception.traceback
+        exception.traceback = (plugin.__module__, lineno, func, msg)
```

Comment:
Don't forget `os.path.abspath(plugin.__module__)` to account for when paths are relative your working directory.

E.g.

1. `cd c:\some\path`
2. `publish()`, for which plugin is at `c:\some\path\plugins\my_plugin.py`
3. Path then becomes `plugins\my_plugin.py`

Reply:
Good call, thanks, it's fixed